app/controllers/api/users_controller.rb

```diff
@@ -54,7 +54,15 @@ def update
         render 'api/errors', locals: { errors: }, status: :unprocessable_entity and return
       end
 
-      if user.update(user_params)
+      activated = user_params[:activated]
+      if activated.nil?
+        update_params = { **user_params }
```

Comment:
ここ、なんで ** してるんですっけ。敢えて複製している? 単に参照でもよさそう。
```suggestion
        update_params = user_params
```

Reply:
型が違うから揃えた方が良いかと思って複製しましたが余計な処理ですね。参照するので問題ないと思うので修正します

Reply:
ActionController::Parameters でしたっけ。たしかに型は違うのですが、 `#update` が求めているのは Hash-like なオブジェクトなので、形を揃えることにあまり意味はないですし、単に参照でよいとおもいます。

※ 型として表現すると `ActionController::Parameters | Hash` で ok なのです。

必要ないところで変換をしていると、なにか意味があるんだっけ? ここの copy は shallow copy でいいの? とか、余計な話に広がってしまい、コードが読みづらくなってしまうので、必要ないコードは書かないほうがハッピーかな、と思っています。